internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -242,41 +255,137 @@ func (srv *Server) updateRuleRoute(w http.ResponseWriter, r *http.Request) {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var changelog []models.BounceRule
+
+		bounce_id := chi.URLParam(r, "bounce_id")
+
+		bouncd_idInt, err := strconv.Atoi(bounce_id)
+
+		if err != nil {
+			log.Println(err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		queryParams := r.URL.Query()
+
+		limit, ok := queryParams["limit"]
 
-func (srv *Server) GetChangelog(w http.ResponseWriter, r *http.Request) {
-	rules, err := srv.DBClient.GetAllChangelogEntries()
+		if !ok {
+			changelog, err = srv.DBClient.GetChangeLogEntries(bouncd_idInt, nil)
+		} else {
+
+			limitAsString := strings.Join(limit, "")
+			limitAsInt, err := strconv.Atoi(limitAsString)
+
+			if err != nil {
+				log.Println(err)
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+
+			changelog, err = srv.DBClient.GetChangeLogEntries(bouncd_idInt, &limitAsInt)
+
+			if err != nil {
+				log.Println(err)
+				http.Error(w, err.Error(), http.StatusNotFound)
+				return
+			}
+
+		}
+
+		ctx := context.WithValue(r.Context(), "changelog", changelog)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+func (srv *Server) GetChangeLogEntriesRoute(w http.ResponseWriter, r *http.Request) {
+	changelog, ok := r.Context().Value("changelog").([]models.BounceRule)
+
+	if !ok {
+		log.Println("ContextValue of rule in GetChangeLogEntriesRoute: " + strconv.FormatBool(ok))
+		paramError := errors.New("Route Parameters")
+		http.Error(w, paramError.Error(), http.StatusBadRequest)
+		return
+	}
+
+	data, err := json.Marshal(changelog)
+
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(data)
+}
+
+func (srv *Server) GetAllChangelogEntries(w http.ResponseWriter, r *http.Request) {
+	changelog, err := srv.DBClient.GetAllChangelogEntries()
 
 	if err != nil {
 		log.Println(err)
 		http.Error(w, err.Error(), http.StatusNotFound)
 		return
 	}
 
-	data, err := json.Marshal(rules)
+	data, err := json.Marshal(changelog)
 
 	if err != nil {
 		log.Println(err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
 	w.Write(data)
 }
 
+// createChangeLogEntryRoute will only be used to seed the change log databse
+// with all current rules
+func (srv *Server) createChangeLogEntryRoute(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	var changelog models.BounceRule
+
+	err := decoder.Decode(&changelog)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = srv.DBClient.CreateChangeLogEntry(changelog.ID, &changelog)
+
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
What happens on success? What StatusCode and response do we send back to the client?